### PR TITLE
First Iteration Proof of concept

### PR DESCRIPTION
**Pull request by rw-wmp**

### CODE REVIEW

**Comment by mkosovec** [comment]
Hey Reid,

I haven't reviewed the code itself yet, but this excel code should not be within the dataops-salesforce folder path.

Please create your own path named dataops-excel at the root level with an identical subfolder structure layout to the dataops-salesforce project/folder.

You can do this by copy-pasting-renaming your existing dataops-salesforce folder locally and then cleaning up both the new and old folder paths.

Your git repository should have two top level folders:

- dataops-salesforce
    - This should only contain information specific to the salesforce plugin, and should only have readme edits in the pull request at most
- dataops-excel
    - This should have all your new code in the same folder/project structure as the dataops-salesforce project, and none of the code from the dataops-salesforce project

Let me know if you have any questions.

Matt

File: dataops-salesforce/README.md [hunk 1]
```diff
@@ -51,4 +68,39 @@ For those who just want to use this as-is, here are the setup steps:
     
 7. Pull now and debug any issues
 
-Please contact your project team for managed service support provider for any issues or assistance debugging the setup or ongoing execution of this custom ingestion codebase
\ No newline at end of file
+Please contact your project team for managed service support provider for any issues or assistance debugging the setup or ongoing execution of this custom ingestion codebase
+
+### **DataOps Excel Plugin**
```
Comment (mkosovec) [inline dataops-salesforce/README.md]: This shouldn't be in this readme.

You should have your own readme in the new dataops-excel top level folder

File: dataops-salesforce/src/main/scala/notebook/dataops-excel.scala [hunk 2]
```diff
@@ -0,0 +1,109 @@
+import org.apache.spark.sql.{DataFrame,SparkSession}
```
Comment (mkosovec) [inline dataops-salesforce/src/main/scala/notebook/dataops-excel.scala]: This file shouldn't be in this location.

The folder path for this file should be:
dataops-excel/src/main/scala/notebook/dataops-excel.scala

**Review by mkosovec — CHANGES_REQUESTED** [review]

File: dataops-salesforce/README.md (hunk 1 above)
Comment (rw-wmp) [reply dataops-salesforce/README.md]: Hi Matt, I copied over and changed the folder structure to match what you recommended. One question I have is around files like dataops-excel/project/target/streams/compile/exportedProducts/_global/streams/export that has references to stuff on your machine - is there a way for me to compile to update those for the excel? Right now im using intelliJ as a staging area for development before copying it into databricks.

Comment (mkosovec) [reply dataops-salesforce/README.md]: Hey Reid,

We should be using the ,gitignore file to exclude any files from the target folder. I have examples in that for the salesforce project.

Anything in the target folder is generated by the compiler, and should not be pushed/stored in git. These files would be re-created by SBT once you open/compile the project.

Comment (mkosovec) [reply dataops-salesforce/README.md]: Annnnd I just realized that those files snuck into this repo with an early push - despite them being in the .gitignore file.

I'm going to add a commit to master to delete that folder from this repo.

Comment (rw-wmp) [reply dataops-salesforce/README.md]: Okay - I just updated all the information in my PR

**Review by mkosovec — APPROVED** [review]

**Comment by mkosovec** [comment]
Ok - let's get this merged in and then we can create a new PR if we have further tweaks

**Pull request merged**